bodhi_messages/bodhi_messages/schemas/stack.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (C) 2018  Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""
+Message schema for Bodhi.
+
+Each message is defined as a Python class. For details, see `fedora-messaging
+<https://fedora-messaging.readthedocs.io/en/stable/>`_ documentation on
+messages.
+"""
+
+from .base import BodhiMessage
+
+
+class StackMessage(BodhiMessage):
+    """Base class for stack messages."""
+
+    @property
+    def summary(self):
+        """
+        A short, human-readable representation of this message.
+
+        This should provide a short summary of the message, much like the subject line
+        of an email.
+        """
+        return self._summary_tmpl.format(
+            agent=self._body['agent'],
+            name=self._body['stack']['name'],
+        )
+
+    @property
+    def url(self):
+        """
+        An URL to the action that caused this message to be emitted.
+
+        Returns:
+            str or None: A relevant URL.
+        """
+        return "https://bodhi.fedoraproject.org/stacks/{title}".format(
+            title=self._body['stack']['name']
+        )
+
+    @property
+    def packages(self):
+        """List of packages affected by the action that generated this message.
+
+        Returns:
+            list(str): A list of affected package names.
+        """
+        return list(set([p['name'] for p in self._body['stack']['packages']]))
+
+
+class StackSaveV1(StackMessage):
+    """Sent when a stack is saved."""
+
+    topic = "bodhi.stack.save"
+    _summary_tmpl = "{agent} updated the \"{name}\" stack"
+
+
+class StackDeleteV1(StackMessage):
+    """Sent when a stack is deleted."""
+
+    topic = "bodhi.stack.delete"
+    _summary_tmpl = "{agent} deleted the \"{name}\" stack"
```

Comment:
I'm sad to say that I'm actually planning on [dropping stacks](https://github.com/fedora-infra/bodhi/issues/2241) as part of [bodhi 4.0.0](https://github.com/fedora-infra/bodhi/projects/6). That said, no harm in accepting this change, it's just sad because I'll be removing it really soon.

Reply:
Stacks is actually dropped now, so I will drop this code before merging.